src/mono/wasm/debugger/DebuggerTestSuite/EvaluateOnCallFrame2Tests.cs

```diff
@@ -711,5 +711,17 @@ public async Task EvaluateLocalObjectFromAssemblyNotRelatedButLoaded()
                    ("EvaluateStaticGetterInValueType.A", TNumber(5))
                 );
            });
+
+        [Fact]
+        public async Task EvaluateSumBetweenObjectAndString() => await CheckInspectLocalsAtBreakpointSite(
+             $"DebuggerTests.SumObjectAndString", "run", 3, "DebuggerTests.SumObjectAndString.run",
+            $"window.setTimeout(function() {{ invoke_static_method ('[debugger-test] DebuggerTests.SumObjectAndString:run'); 1 }})",
+            wait_for_event_fn: async (pause_location) =>
+            {
+                var id = pause_location["callFrames"][0]["callFrameId"].Value<string>();
+                await EvaluateOnCallFrameAndCheck(id,
+                   ("myList+\"asd\"", TString("System.Collections.Generic.List`1[System.Int32]asd"))
```

Comment:
- can we also test this for a type that has `ToString()` overridden?
- And test `object o = new()`
- And what happens in case of a value type?
- what would happen for `list l = null; l + "asd"` ?

What about these?:
- object/vt + object/vt?
- object/vt + non-string primitive type?
- Any other binary operators, what happens in those cases?
These don't need to be fixed in this PR, I'm just curious if they would work or if they will at least fail without crashing.